hack/test-go.sh

```diff
@@ -131,7 +131,8 @@ junitFilenamePrefix() {
     return
   fi
   mkdir -p "${KUBE_JUNIT_REPORT_DIR}"
-  echo "${KUBE_JUNIT_REPORT_DIR}/junit_${KUBE_API_VERSION}_$(kube::util::sortable_date)"
+  local KUBE_TEST_API_NO_SLASH=echo "${KUBE_TEST_API//\//-}"
```

Comment:
please add a comment describing the expected format of this variable-- I'm worried there will be collisions. Aren't there multiple groups/versions in that variable now?